Pr merge policies

chrisstafford opened the pull request:
---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1201794020676141

chrisstafford committed:
adds pull request approval process

joachimweyl reviewed (approved):
Looks good.

larsks commented:
This isn't a policy so much as it is a "how to" document. I think we need to iterate on it a bit before merging.

larsks reviewed (changes requested):
See my comment re: policy vs how-to

joachimweyl commented:
@mikthoma do you feel that you can step in and make the changes necessary to format this as a policy rather than a How-to?

larsks commented:
@joachimweyl I think the policy side might need to come from us.

joachimweyl commented:
@mikthoma ignore that request.

joachimweyl commented:
@jtriley do you have a set idea of PR merge policies that you could provide here?

jtriley commented:
This looks good as a contributing guide. I suppose a good initial policy for repos is enforcing protected branch for main, require ci/cd to pass, requiring two reviews, and all conversations to be resolved before merge.

jtriley commented:
For CI/CD that deploys to a production system we should consider end user impact of the changes and schedule a maintenance event if necessary before merging.

joachimweyl committed:
Merging Original with justin's comments

The original was more of a how to than a policy.
Merged parts of how to and justin's comments to create a policy.

joachimweyl commented:
@larsks I have merged the policies from @chrisstafford's how-to and comments from @jtriley to create a policy list. Please let me know if this was more like what you had in mind.

knikolla commented on pull_request_approval_policy.md:
```diff
@@ -0,0 +1,10 @@
+# Pull Request Approval Policies
+
+- Enforce protected branch for main
```
The first 3 lines aren't really approval policies for a PR as much as they are guidelines for setting up repos.

joachimweyl replied:
That makes sense. Thank you.
Updated.

joachimweyl committed:
Paired down to be just PR policies

some items were more guidelines for setting up repos than
PR policies, removed them.

larsks commented on pull_request_approval_policy.md:
```diff
@@ -0,0 +1,7 @@
+- CI/CD is required to pass
+- 2 Reviewers are required for all Pull Requests
+- All Change Requests must be resolved prior to merging
+- For CI/CD that deploy to a production system schedule a maintenance event if
```
I think this last statement is a little wishy-washy.  How about:

- Changes that impact service availability must only be merged during a pre-scheduled maintenance window.

joachimweyl replied:
updated.

joachimweyl committed:
Reword policy to be less wishy-washy

joachimweyl commented:
This is the set of Policies for the creation of this project. NERC may choose to make changes to this once they are managing the PRs fully. At that time this doc might need to be updated.
